### PR TITLE
[CPU][ARM] JIT Ceiling Operation

### PR DESCRIPTION
**Pull request by xyz-harshal**
### Details:
 - Added JIT emitter for Eltwise Ceiling operation on ARM64 SIMD
 - Implemented fp32 optimization replacing C++ Math implementation
 - Modified ARM64 executor to support new JIT emitter
 - Updated kernel files to include Ceiling in Eltwise operations
 - Added test coverage for JIT implementation verification
 - Transitioned operation type from Math to Eltwise for better performance

### Tickets:
 - #27498

### CODE REVIEW

**Comment by xyz-harshal** [comment]
image
## This instances passes all the test locally
The image mentioned above shows that all the test have passed.
## Description

I actually forgot to add a line in the [eltwise.cpp](https://github.com/xyz-harshal/openvino/blob/master/src/plugins/intel_cpu/src/nodes/eltwise.cpp)

Hopefully all the test cases on github should pass!

## Reviewers

@a-sidorova , @ilya-lavrenov , @dmitry-gorokhov  - Please review the changes made in this PR.

Thank you!

**Comment by a-sidorova** [comment]
@xyz-harshal Hi! Thanks a lot for the contribution!

I observe some failed tests:
```
[ RUN      ] smoke_Activation4D_Snippets_CPU/ActivationLayerCPUTest.CompareWithRefs/Ceiling_TS=(2.17.5.4)_AS=()_ConstantsValue=()_netPRC=f32_inPRC=f32_outPRC=f32__inFmts=nhwc_outFmts=nhwc_enforceSnippets=1

MEM_USAGE=2253716KB
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:97: Failure
Exception from src/inference/src/cpp/core.cpp:104:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'jitter != jitters.end()' failed at src/common/snippets/src/lowered/target_machine.cpp:19:
Supported precisions set is not available for Ceiling operation.



[  FAILED  ] smoke_Activation4D_Snippets_CPU/ActivationLayerCPUTest.CompareWithRefs/Ceiling_TS=(2.17.5.4)_AS=()_ConstantsValue=()_netPRC=f32_inPRC=f32_outPRC=f32__inFmts=nhwc_outFmts=nhwc_enforceSnippets=1, where GetParam() = ({ ({}, { { 2, 17, 5, 4 } }) }, {}, (29, {}), f32, f32, f32, ({ 7 }, { 7 }, {}, ""), true) (29 ms)
```

Could you please remove your addition of `Ceiling` to `activationTypesSnippets` [there](https://github.com/openvinotoolkit/openvino/blob/e1d8edd0542430da8cdb190683affea681c819b6/src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/activation.cpp#L273)? `Ceiling` op is not supported by Snippets for now, so no need to add this op to the validation list of types for Snippets.

**Comment by xyz-harshal** [comment]
Hi @a-sidorova i have made changes in the file as you said.
Can you start the CI?

File: src/plugins/intel_cpu/src/nodes/eltwise.cpp [hunk 1]
```diff
@@ -257,6 +257,7 @@ std::set<std::vector<element::Type>> eltwise_precision_helper::get_supported_pre
         OV_CASE(Algorithm::EltwiseMultiply, jit_multiply_emitter),
         OV_CASE(Algorithm::EltwiseDivide, jit_divide_emitter),
         OV_CASE(Algorithm::EltwiseFloor, jit_floor_emitter),
+        OV_CASE(Algorithm::EltwiseCeiling,jit_ceiling_emitter),
```
[the remark this answers is not in the document]
Comment (a-sidorova) [reply src/plugins/intel_cpu/src/nodes/eltwise.cpp]: L640 has not been changed :)

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp [hunk 2]
```diff
@@ -516,6 +516,50 @@ std::set<std::vector<element::Type>> jit_floor_emitter::get_supported_precisions
     return {{element::f32}};
 }
 
+/// CEILING ///
+//Initialization of the emitter, taking node as input
+jit_ceiling_emitter::jit_ceiling_emitter(dnnl::impl::cpu::aarch64::jit_generator* host,
+                                     dnnl::impl::cpu::aarch64::cpu_isa_t host_isa,
+                                     const std::shared_ptr<ov::Node>& node)
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp:523]: Minor: fix please the tabulation here and on L529-530

**Comment by xyz-harshal** [comment]
Hi @a-sidorova i have made changes in the file as you said.
Can you start the CI?

**Comment by a-sidorova** [comment]
build_jenkins

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp [hunk 3]
```diff
@@ -312,6 +312,9 @@ std::vector<std::string> disabledTestPatterns() {
     // fp32 floor for bf16 models: conversion issue
     retVector.emplace_back(R"(.*smoke.*ActivationLayerCPUTest.*CompareWithRefs/Floor_.*netPRC=bf16.*)");
 
+    // Issue: #27498
+    retVector.emplace_back(R"(.*smoke.*ActivationLayerCPUTest.*CompareWithRefs/Ceiling_.*netPRC=bf16.*)");
+
```
[the remark this answers is not in the document]
Comment (a-sidorova) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp]: After returning these tests, they're [failed](https://github.com/openvinotoolkit/openvino/actions/runs/11947374551/job/33303919436?pr=27527) as `Floor`:
```
[ RUN      ] smoke_Activation4D_Eltwise_CPU_Blocked/ActivationLayerCPUTest.CompareWithRefs/Ceiling_TS=(2.4.4.1)_AS=()_ConstantsValue=()_netPRC=bf16_inPRC=f32_outPRC=f32__inFmts=nChw16c_outFmts=nChw16c_enforceSnippets=0

MEM_USAGE=1809868KB

Expected: 13 Actual: 12 Coordinate: 20 Diff: 1 calculated_abs_threshold: 0.19 abs_threshold: 0.06 rel_threshold: 0.01
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:97: Failure
[ COMPARATION ] COMPARATION IS FAILED! incorrect elem counter: 1 among 32 shapes.
[  FAILED  ] smoke_Activation4D_Eltwise_CPU_Blocked/ActivationLayerCPUTest.CompareWithRefs/Ceiling_TS=(2.4.4.1)_AS=()_ConstantsValue=()_netPRC=bf16_inPRC=f32_outPRC=f32__inFmts=nChw16c_outFmts=nChw16c_enforceSnippets=0, where GetParam() = ({ ({}, { { 2, 4, 4, 1 } }) }, {}, (29, {}), bf16, f32, f32, ({ 9 }, { 9 }, {}, ""), false) (27 ms)
```

So I suggest to do the same as for `Floor` op: just skip these tests with `Ceiling` if `netPrc=bf16`.
Could you please update the line [L313](https://github.com/openvinotoolkit/openvino/blob/e1d8edd0542430da8cdb190683affea681c819b6/src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp#L313C4-L313C105) that skips `Floor` for bf16 so ?
` retVector.emplace_back(R"(.*smoke.*ActivationLayerCPUTest.*CompareWithRefs/(Floor|Ceiling)_.*netPRC=bf16.*)");`

**Comment by a-sidorova** [comment]
build_jenkins

**Review by a-sidorova — APPROVED** [review]
@xyz-harshal Thank you for the contribution! 👍🏼

**Pull request merged**